scripts/linters/pylint_extensions.py

```diff
@@ -1484,31 +1484,77 @@ class RestrictedImportChecker(checkers.BaseChecker):
+                    'that are forbidden separated by \'|\', these imports are '
+                    'relative to the root module (e.g. \'core.domain\').'
+                )
+            }
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(RestrictedImportChecker, self).__init__(linter)
+        self._module_to_forbidden_imports = []
+
+    def open(self):
+        """Parse the forbidden imports."""
+        splitted_module_to_forbidden_imports = [
+            forbidden_import.strip().split(':')
+            for forbidden_import in self.config.forbidden_imports
+        ]
+        self._module_to_forbidden_imports = list(
+            (
+                forbidden_imports[0].strip(),
+                [import_.strip() for import_ in forbidden_imports[1].split('|')]
+            ) for forbidden_imports in splitted_module_to_forbidden_imports
+        )
+
+    def _iterate_forbidden_imports(self, node):
+        """Yields pairs of module name and forbidden imports.
+
+        Args:
+            node: astroid.node_classes.Import. Node for a import statement
+                in the AST.
+
+        Yields:
+            tuple(str, str). Yields pair of module name and forbidden import.
+        """
+        modnode = node.root()
+        for module_name, forbidden_imports in self._module_to_forbidden_imports:
+            for forbidden_import in forbidden_imports:
+                if module_name in modnode.name and not '_test' in modnode.name:
+                    yield module_name, forbidden_import
+
     def visit_import(self, node):
         """Visits every import statement in the file.
 
         Args:
             node: astroid.node_classes.Import. Node for a import statement
                 in the AST.
         """
-
-        modnode = node.root()
         names = [name for name, _ in node.names]
-        # Checks import of domain layer in storage layer.
-        if 'oppia.core.storage' in modnode.name and not '_test' in modnode.name:
-            if any('core.domain' in name for name in names):
-                self.add_message(
-                    'invalid-import',
-                    node=node,
-                    args=('domain', 'storage'),
-                )
-        # Checks import of controller layer in domain layer.
-        if 'oppia.core.domain' in modnode.name and not '_test' in modnode.name:
-            if any('core.controllers' in name for name in names):
+        for module_name, forbidden_import in self._iterate_forbidden_imports(
+                node):
+            if any(forbidden_import in name for name in names):
                 self.add_message(
                     'invalid-import',
                     node=node,
-                    args=('controller', 'domain'),
+                    args=(
+                        forbidden_import.split('.')[-1],
+                        module_name.split('.')[-1]
+                    ),
```

Comment:
line 1548-1557 is still a duplicate of the lines 1568-1577? Can we have one _check.... function and pass list of module names as a paramenetr?

Reply:
They are not the same, there is and different condition.

Reply:
But I moved other parts of the code to separate function.